### PR TITLE
Refactor db methods, simplify header generation, fix caching

### PR DESCRIPTION
**Pull request by toabctl**

### CODE REVIEW

**Comment by toabctl** [comment]
cc @Oneiroi

**Comment by toabctl** [comment]
@Oneiroi any comments here?

**Comment by toabctl** [comment]
@Oneiroi btw. travis seems to be disabled still

**Comment by Oneiroi** [comment]
Indeed, have some issues to resolve prior to enablement as there is no way
to separate org permissions and individual permission for Travis it seems,
working on this.

Typed from my cellphone, please forgive any typos and/or grammatical errors

**Comment by toabctl** [comment]
> Indeed, have some issues to resolve prior to enablement as there is no way to separate org permissions and individual permission for Travis it seems, working on this.

Sure. But you can also run the testsuite manually and review the code. We don't need travis get this merged :)

**Comment by Oneiroi** [comment]
Indeed, as soon as I get a moment I will do so and merge, likely this
evening post $work, thanks!

**Comment by toabctl** [comment]
> Indeed, as soon as I get a moment I will do so and merge, likely this evening post $work, thanks!

@Oneiroi ok. btw. are you still using this script at percona for health check? or is there something else you use?

**Comment by Oneiroi** [comment]
@toabctl `percona` pulls (iirc) the version from `master` hence my comments on #21 the `twisted` branch version is more performant as multi-mechanise benchmarks show; however there was issues with the twisted versions being available to some distros that would mean the install was not simple and required extra steps.

**Comment by toabctl** [comment]
@Oneiroi friendly ping for a review

**Comment by toabctl** [comment]
@Oneiroi another friendly ping for review

**Comment by Oneiroi** [comment]
Apologies,

OoO will review on return, Travis CICD was enabled, need to ensure
functional however.

**Comment by toabctl** [comment]
@Oneiroi friendly ping for a review

**Comment by toabctl** [comment]
@Oneiroi another friendly ping for a review. Anything I can help with here?

File: README.md [hunk 1]
```diff
@@ -10,9 +10,23 @@ This version is rewritten to use Python twisted, so this module is required.
 To start at run time:
 echo "/usr/bin/pyclustercheck -f /etc/my.cnf > /var/log/messages 2>&1 &" >> /etc/rc.local
```
Comment (Oneiroi) [inline README.md]: No related to this PR, just notice this line is dangerous and would wipe out /var/log/messages on run; this will need correction.

**Comment by Oneiroi** [comment]
Apologies @toabctl my "spare time" is non existent at this moment in time, have you run the multimechanize benchmarks for this change, does the performance hold up ?

File: README.md (hunk 1 above)
Comment (toabctl) [reply README.md]: pyclustercheck is also not the script name :)

**Comment by toabctl** [comment]
> Apologies @toabctl my "spare time" is non existent at this moment in time, have you run the multimechanize benchmarks for this change, does the performance hold up ?

I have not done the benchmarks yet. Let me check if I'm able to do that.

**Comment by toabctl** [comment]
@Oneiroi one more question about performance - why does performance matter for this script? In our case, we just do a single request per second from HAProxy. So why does it matter? How are you using this script?

**Comment by Oneiroi** [comment]
@toabctl In history this use SimpleHTTPServer (this is the version in `master`) this suffered stalls which could not be resolved without restarting the service (which sometimes became true in the `xinetd` version also iirc); I setup the multimech benchmarks to show the improvement of the twisted version several years ago, and this also demonstrated that there were no longer any stalls occurring.

No one use case is the same, the benchmarks are more to check for such hang ups.

**Comment by toabctl** [comment]
@Oneiroi given that I implemented the systemd watchdog support, the "do not stall" is no longer that relevant because the service would be restarted anyway.

**Comment by Oneiroi** [comment]
@toabctl with that in consideration I have completed review, and merged the changes, I see nothing from quick review that could introduce issues to a system running this; as such I am ok for this merge to take please, many thanks for your contribution!

**Comment by toabctl** [comment]
@Oneiroi I wonder if we could get rid of twisted again. It pulls in a lot of dependencies...

**Comment by toabctl** [comment]
@Oneiroi with python3, I think twisted could be replaced with async . but I have not used it yet. Anyway - thanks for reviewing it and helping out!

**Pull request merged**